src/WinRT.Runtime/ObjectReference.ReferenceTracking.cs

```diff
@@ -0,0 +1,240 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Threading;
+
+namespace WinRT
+{
+    partial class IObjectReference
+    {
+        /// <summary>
+        /// Finalizes the current object instance. If the underlying native
+        /// resources are still active, it also releases them as needed.
+        /// </summary>
+        ~IObjectReference()
+        {
+            DisposeUnsafe();
+        }
+
+        /// <inheritdoc/>
+        public void Dispose()
+        {
+            GC.SuppressFinalize(this);
+
+            DisposeUnsafe();
+        }
+
+        /// <summary>
+        /// Gets the underlying pointer owned by the current instance.
+        /// </summary>
+        /// <returns>The underlying pointer owned by the current instance.</returns>
+        /// <remarks>
+        /// This method does not check for disposal, nor does it increment the managed reference count of
+        /// the current object. Callers must call <see cref="AddRefUnsafe"/> and <see cref="ReleaseUnsafe"/>.
+        /// </remarks>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public IntPtr GetRefUnsafe()
```

Comment:
Given that we have apis like Unsafe.AsRef, probably want to use a different name here than "Ref".